pandas/io/common.py

```diff
@@ -194,20 +194,22 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
     """
     filepath_or_buffer = _stringify_path(filepath_or_buffer)
 
+    from io import TextIOWrapper
     if _is_url(filepath_or_buffer):
         req = _urlopen(filepath_or_buffer)
         content_encoding = req.headers.get('Content-Encoding', None)
         if content_encoding == 'gzip':
             # Override compression based on Content-Encoding header
             compression = 'gzip'
         reader = BytesIO(req.read())
-        return reader, encoding, compression
+        return TextIOWrapper(reader, encoding=encoding), encoding, compression
```

Comment:
I think this change is causing problems elsewhere:  https://travis-ci.org/pandas-dev/pandas/jobs/269765182#L1928 Are you able to reproduce that locally?

Reply:
Was analyzing last night. Trying to reproduce locally this morning with
python 2.7. Cannot reproduce with 3.6.

Reply:
I can reproduce with 2.7... looking into it.

Reply:
I made a change to have this revert to previous functionality when not using Python 3. My new tests + the failing tests now pass on 2.7 and 3.6 locally. Resubmitting for a full build.